internal/search/subrepoperms/sub_repo_perms_job.go

```diff
@@ -0,0 +1,116 @@
+package subrepoperms
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/cockroachdb/errors"
+	"github.com/hashicorp/go-multierror"
+	"github.com/inconshreveable/log15"
+
+	"github.com/sourcegraph/sourcegraph/internal/actor"
+	"github.com/sourcegraph/sourcegraph/internal/authz"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/search/result"
+	"github.com/sourcegraph/sourcegraph/internal/search/run"
+	"github.com/sourcegraph/sourcegraph/internal/search/streaming"
+)
+
+// NewSubRepoPermsFilterJob creates a job that filters the streamed results
+// of its child job using the default authz.DefaultSubRepoPermsChecker.
+func NewSubRepoPermsFilterJob(child run.Job) run.Job {
```

Comment:
```suggestion
func NewFilterJob(child run.Job) run.Job {
```

Suggestion here, for destuttering. Or `NewResultFilterJob`. Side note: Not a huge fan of this `subrepoperms` package name or naming around this operation in general but since we've settled on `authz.SubRepoPermissionChecker` I'm OK with it for consistency (the name really sticks out suggesting we don't have a cohesive permissions model)

Reply:
> (the name really sticks out suggesting we don't have a cohesive permissions model)

+1.

Reply:
Agree on the name. Was initially in the `run` package, but I decided I didn't want implementations of jobs cluttering up the package with the definition of a job.

Reply:
yep i agree with that